File: CS/WindowsApplication1/Form1.cs
using System.Windows.Forms;
using DevExpress.XtraEditors;
using System;

namespace WindowsFormsApplication3
{
    public partial class Form1 : XtraForm, IHelpOwner
    {
        HelpHelper helper;
        public Form1()
        {
            this.helper = new HelpHelper(this);
            InitializeComponent();
        }

        private void Form1_HelpRequested(object sender, HelpEventArgs hlpevent)
        {
            Text += "1";
            this.helper.SetHelpRequested();
            Control control = this.GetChildAtPoint(this.PointToClient(hlpevent.MousePos));
            XtraMessageBox.Show(String.Format("Help requested for {0}", control.Name));
        }
        protected override void WndProc(ref Message msg) {
            if (helper.DoWndProc(ref msg)) return;
            base.WndProc(ref msg);
        }
        #region IHelpOwner Members
        void IHelpOwner.DoBaseWndProc(ref Message msg) {
            base.WndProc(ref msg);
        }
        void IHelpOwner.DoHelpRequested() {
            OnHelpRequested(new HelpEventArgs(MousePosition));
        }
        Control IHelpOwner.Control { get { return this; } }

        #endregion
    }
}

File: VB/WindowsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports System

Namespace WindowsFormsApplication3
	Partial Public Class Form1
		Inherits XtraForm
		Implements IHelpOwner
		Private helpHelper As HelpHelper
		Public Sub New()
			Me.helpHelper = New HelpHelper(Me)
			InitializeComponent()
		End Sub

		Private Sub Form1_HelpRequested(ByVal sender As Object, ByVal hlpevent As HelpEventArgs) Handles MyBase.HelpRequested
			Text &= "1"
			Me.helpHelper.SetHelpRequested()
			Dim control As Control = Me.GetChildAtPoint(Me.PointToClient(hlpevent.MousePos))
			XtraMessageBox.Show(String.Format("Help requested for {0}", control.Name))
		End Sub
		Protected Overrides Sub WndProc(ByRef msg As Message)
			If helpHelper.DoWndProc(msg) Then
				Return
			End If
			MyBase.WndProc(msg)
		End Sub
		#Region "IHelpOwner Members"
		Private Sub DoBaseWndProc(ByRef msg As Message) Implements IHelpOwner.DoBaseWndProc
			MyBase.WndProc(msg)
		End Sub
		Private Sub DoHelpRequested() Implements IHelpOwner.DoHelpRequested
			OnHelpRequested(New HelpEventArgs(MousePosition))
		End Sub
		Private ReadOnly Property IHelpOwner_Control() As Control Implements IHelpOwner.Control
			Get
				Return Me
			End Get
		End Property

		#End Region
	End Class
End Namespace
